workflow/Snakefile

```diff
@@ -9,85 +9,84 @@ import yaml
 configfile: "config/config.yaml"
 conda: "mamba"
 
-BIOSAMPLE_DF = pd.read_table(config["ABC_BIOSAMPLES"], na_values="").set_index("biosample", drop=False)
-
-def make_paths_absolute(obj, base_path):
-	"""
-	Use absolute paths to be compatible with github submodules
-	Recursively go through the dictionary and convert relative paths to absolute paths.
-	"""
-	if isinstance(obj, dict):
-		for key, value in obj.items():
-			obj[key] = make_paths_absolute(value, base_path)
-	elif isinstance(obj, str):
-		# We assume all strings are paths. If converting the string
-		# to an absolute path results in a valid file, then the str was a path
-		new_file = os.path.join(base_path, obj)
-		if os.path.exists(new_file):
-			return new_file
-	return obj
 
-config = make_paths_absolute(config, os.getcwd())
+include: os.path.join("rules", "utils.smk")
+
+
+## Update paths in the config to be compatible with github submodules
 # Need to manually make results_dir an absolute path since above may
 # not work if results_dir folder isn't created
 # If results_dir is already an absolute path, this is a no-op
+config = make_paths_absolute(config, os.getcwd())
 config["results_dir"] = os.path.join(os.getcwd(), config["results_dir"])
-RESULTS_DIR = config["results_dir"]
-E2G_DIR_PATH = config["E2G_DIR_PATH"]
-MODEL_DIR = config["model_dir"] # used in utils.smk to get correct model
-include: os.path.join(E2G_DIR_PATH, "workflow", "rules", "utils.smk")
-
-def get_e2g_config(config):
-	e2g_config_file = os.path.join(E2G_DIR_PATH, "config/config.yaml")
-	with open(e2g_config_file, 'r') as stream:
-		e2g_config = yaml.safe_load(stream)
-	e2g_config["E2G_DIR_PATH"] = E2G_DIR_PATH
-	e2g_config["ABC_BIOSAMPLES"] = config["ABC_BIOSAMPLES"]
-	e2g_config["results_dir"] = config["results_dir"]
-	e2g_config["model_dir"] = config["model_dir"]
-	config["MEGAMAP_HIC_FILE"] = e2g_config["MEGAMAP_HIC_FILE"]
-	return e2g_config
 
+
+## Define global variables
+RESULTS_DIR = config["results_dir"] # used in ENCODE_rE2G/ABC/workflow/rules/macs2.smk 
+
+
+## Convert cell cluster config to a biosample config for the ABC pipeline
+## Treat each cell cluster as a distinct biosample
+make_biosample_config(
```

Comment:
Related to multiple experiments:

I think a better way to do this is to make creating the ABC biosamples config as a separate rule.

One issue I see with this approach is that it writes to the tmp folder. This can be an issue when we run multiple experiments. It'd be better if we write the abc_biosamples.tsv to the folder specific to the experiment, e.g results/xu_et_al/abc_biosamples.tsv